src/main.cpp

```diff
@@ -5,30 +5,43 @@
 * Author : Andre
 */
 
-#define SPR (400 * 8)
-#define EARTH_SPEED (7.2921150 / 100000)
-#define TRANSMISSION 35.46611505122143
+
 
 #include <Arduino.h>
 
-#include "FastPin.h"
+#include "hal/avr/AVRFastPin.h"
 #include "PreciseInterrupt.h"
 #include "InterruptStepper.h"
 
+PreciseInterrupt isr1;
+
+static inline __attribute__((always_inline)) void stepRA(uint8_t direction) {
+	/* TODO use direction
+	if (direction) {
+		
+	}
+	else {
+
+	}
+	*/
+	AVRFastPin<54>::pulse();
+}
+
+
 void setup()
 {
-	FastPin<60>::output();
-	FastPin<54>::output();
-	FastPin<38>::output();
-	FastPin<38>::low();
+	AVRFastPin<60>::output();
```

Comment:
I'm unsure exactly how we'll work this all out... but  possibly have setup functions specific to each hardware type, and the setup() / loop() just call those based on environment.